gdal/ogr/ogrct.cpp

```diff
@@ -2399,6 +2442,37 @@ int OGRProjCT::TransformWithErrorCodes(
     return TRUE;
 }
 
+/************************************************************************/
+/*                               Clone()                                */
+/************************************************************************/
+
+OGRCoordinateTransformation* OGRProjCT::Clone() const
+{
+    OGRProjCT* poNewCT = new OGRProjCT(*this);
+#if (PROJ_VERSION_MAJOR * 10000 + PROJ_VERSION_MINOR * 100 + PROJ_VERSION_PATCH) < 80001
+    // See https://github.com/OSGeo/PROJ/pull/2582
+    // This may fail before PROJ 8.0.1 if the m_pj object is a "meta"
+    // operation being a set of real operations
+    bool bCloneDone = ((m_pj == nullptr) == (poNewCT->m_pj == nullptr));
```

Comment:
this check is wrong. You need to go through the Initialize() road if poNewCT->m_pj == nullptr ad m_pj != nullptr. And I don't think you don't need to iterate over m_oTransformations

Reply:
Looks like I misunderstood your note "proj_clone might fail in some circumstances for PROJ < 8.0.1" from [here](https://github.com/OSGeo/gdal/pull/3809#discussion_r627754095).

My logic was based on these statements:

* `proj_clone` return null on failure
* If `this` `OGRProjCT` object contains non-null `m_pj` value and `poNewCT->m_pj` is non-null object I'm expect that `poNewCT->m_pj` is valid and nothing to do. 
* If `this` `OGRProjCT` object contains null `m_pj` value and `poNewCT->m_pj` is null object I'm expect that null is valid for such case and nothing to do. 
* Other "special" values (bWebMercatorToWGS84LongLat, bNoTransform, etc) are copied at constructor and we don't need to recompute it unlike `GetInverse`.

Reply:
you're right. I was too quick on saying that " ((m_pj == nullptr) == (poNewCT->m_pj == nullptr))" was wrong. Was just confused by the equality of equalities :-). The new code looks right AFAICS